packages/@aws-cdk/aws-sagemaker/test/test.endpoint.ts

```diff
@@ -0,0 +1,170 @@
+import { expect, haveResource } from '@aws-cdk/assert';
+import ec2 = require('@aws-cdk/aws-ec2');
+import kms = require('@aws-cdk/aws-kms');
+import cdk = require('@aws-cdk/core');
+import { Test } from 'nodeunit';
+import sagemaker = require('../lib');
+
+export = {
+    "When creating Sagemaker Endpoint": {
+        "with only required properties set, it correctly sets default properties"(test: Test) {
+            // GIVEN
+            const stack = new cdk.Stack();
+            const model = sagemaker.Model.fromModelName(stack, 'Model', "mymodel");
+
+            // WHEN
+            new sagemaker.Endpoint(stack, 'Endpoint', {
+                productionVariants: [ { model } ],
+            });
+
+            // THEN
+            expect(stack).to(haveResource("AWS::SageMaker::EndpointConfig", {
```

Comment:
General comment for all tests: we have also a `haveResourceLike` matcher that allows you to check only the parts of the resource that you're interested in in a given test. Might come in handy for these.